src/sql/workbench/services/notebook/browser/notebookServiceImpl.ts

```diff
@@ -286,7 +294,17 @@ export class NotebookService extends Disposable implements INotebookService {
 				}
 			}
 		}
-		return await this._editorService.openEditor(fileInput, editorOptions, viewColumnToEditorGroup(this._editorGroupService, options.position));
+
+		return fileInput;
+	}
+
+	public async openNotebook(resource: UriComponents, options: INotebookShowOptions): Promise<IEditorPane | undefined> {
+		const editorOptions: ITextEditorOptions = {
+			preserveFocus: options.preserveFocus,
+			pinned: !options.preview
+		};
+		let input = await this.createNotebookInput(options, resource);
```

Comment:
Is this going to handle an undefined input correctly?

(that's assuming that having an undefined input is even a valid scenario, which I asked about in a different comment)

Reply:
Fixed by changing it to throw an error instead of returning undefined.